app/Actions/Todo/StoreTodoAction.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Actions\Todo;
+
+use App\Models\Todo;
+use App\Repositories\Interfaces\TodoRepositoryInterface;
+use Illuminate\Database\Eloquent\Model;
+
+class StoreTodoAction
+{
+    private TodoRepositoryInterface $todoRepository;
+
+    public function __construct(TodoRepositoryInterface $todoRepository)
+    {
+        $this->todoRepository = $todoRepository;
```

Comment:
read the [article](https://stitcher.io/blog/constructor-promotion-in-php-8) and replace here and further